File: 07-BinarySearch/BinarySearch.py
# binary search algorithim

# must declare function before using it 
def binarySearch(array, target):
  high = len(array) -1
  low = 0

  while low <= high:
    mid = (high+low) // 2 #use integer division
    if array[mid] < target:
      low = mid + 1
    elif array[mid] > target:
      high = mid - 1
    elif array[mid] == target:
      return mid
    
  return -1 # return -1 if not found 

nums = [1, 3, 5, 7, 8, 10]
index = binarySearch(nums, 10)
print(f"The number 10 is found at index: {index}")

File: 07-BinarySearch/BinarySearch.js
// binary search algorithim
// run in terminal using node BinarySearch.js

const nums = [1, 3, 5, 7, 8, 10];
const index = binarySearch(nums, 10);
console.log("The number 10 is found at index: " + index);


function binarySearch(array, target) {
  let high = array.length - 1;
  let low = 0;

  while (low <= high) {
    let mid = Math.floor((high + low) / 2); // js does not truncate, need to use math.floor
    if (array[mid] < target) {
      low = mid + 1;
    } else if (array[mid] > target) {
      high = mid - 1;
    } else if (array[mid] == target) {
      return mid;
    }
  }
  return -1; // return -1 if not found
}